### PR TITLE
Add testing infra. and first unit-test for e-cash recovery 

### PR DESCRIPTION
**Pull request by dpc**
Follow-up to #1015 

Add some basic infra to allow reasonable code to test recovery logic, and a first test to validate the approach.

Also, store unsigned notes in the backup and add code to handle them during recovery.

### CODE REVIEW

**Comment by dpc** [comment]
@elsirion Please let me know if this fundamentally looks good to you, so I can write bunch of other tests using similar approach.

File: client/client-lib/src/mint/backup.rs [hunk 1]
```diff
@@ -403,7 +417,25 @@ impl EcashRecoveryNonceTracker {
                 .map(|(amount, note)| (note.note.0, (amount, note)))
                 .collect(),
             // TODO: needs to get these in a backup as well
```
Comment (elsirion) [inline client/client-lib/src/mint/backup.rs:419]: Is this TODO fixed now?

File: client/client-lib/src/mint/backup/tests.rs [hunk 2]
```diff
@@ -1,6 +1,236 @@
...
+
+                                (
+                                    amount,
+                                    (
+                                        blind_message,
+                                        tbs::sign_blinded_msg(
+                                            blind_message,
+                                            *sec_keys
+                                                .get(amount)
+                                                .expect("key for amount must be there"),
```
Comment (elsirion) [inline client/client-lib/src/mint/backup/tests.rs:175]: Master of nesting :laughing: (it's only test code, so :man_shrugging: )

**Review by elsirion — APPROVED** [review]

File: client/client-lib/src/mint/backup.rs (hunk 1 above)
Comment (elsirion) [reply client/client-lib/src/mint/backup.rs:419]: Can be fixed later

**Pull request merged**